File: testprojects/java/benchmark/accuracy-test-2013-04-17/domain/indirect/allowedfrom/AccessInstanceVariableIndirectIndirect_MethodVarVar.java
package domain.indirect.allowedfrom;

import domain.indirect.intermediate.BackgroundService;

public class AccessInstanceVariableIndirectIndirect_MethodVarVar {
	private BackgroundService bckg;
	private String test;

	public AccessInstanceVariableIndirectIndirect_MethodVarVar(){
		bckg = new BackgroundService();
	}
		// Indirect-indirect access of a normal attribute 
		public String case4() {
			test = bckg.getServiceTwo().serviceOne.name;
			return test;
		}
}		

File: src/test/resources/csharp/benchmark/Domain/Indirect/ViolatingFrom/AccessInstanceVariableIndirectIndirect_MethodVarVar.cs
namespace Domain.Indirect.ViolatingFrom
{
	public class AccessInstanceVariableIndirectIndirect_MethodVarVar
	{
        private Domain.Indirect.Intermediate.BackgroundService bckg;

        private string test;

        public void Case4()
        {
            bckg = new Domain.Indirect.Intermediate.BackgroundService();
            test = bckg.GetServiceTwo().serviceOne.name;
        }
	}
}
